modules/core/02-client/migrations/v7/genesis.go

```diff
@@ -0,0 +1,82 @@
+package v7
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+// MigrateGenesis accepts an exported IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v2 to v3)
+// - Remove all solo machine consensus states
+// - Remove localhost client
+func MigrateGenesis(clientGenState *types.GenesisState, cdc codec.ProtoCodecMarshaler) (*types.GenesisState, error) {
+	// To prune the client and consensus states, we will create new slices to fill up
+	// with information we want to keep.
+	var (
+		clientsConsensus []types.ClientConsensusStates
+		clients          []types.IdentifiedClientState
+	)
+
+	for _, client := range clientGenState.Clients {
+		clientType, _, err := types.ParseClientIdentifier(client.ClientId)
+		if err != nil {
+			return nil, err
+		}
+
+		// update solo machine client state defintions
```

Comment:
[nit] should we move this comment inside the solomachine case? The way we do for the localhost case.

Reply:
I removed the comment. I don't think it added much information